cli-tests/tests/pmm-cli/server/install.spec.ts

```diff
@@ -5,57 +5,60 @@ import { teardown } from '@tests/helpers/containers'
 
 test.describe.configure({ mode: 'parallel' })
 
-test.describe('Install PMM Server', async () => {
-  test('shall respect relevant flags', async ({ }) => {
-    const adminPw = 'admin123'
-    const containerName = 'pmm-server-install-test'
-    const imageName = 'percona/pmm-server:2.32.0'
-    const volumeName = 'pmm-data-install-test'
-
-    try {
-      let output = await cli.exec(`
-        pmm server docker install 
-          --json
-          --admin-password=${adminPw}
-          --docker-image="${imageName}"
-          --https-listen-port=1443
-          --http-listen-port=1080
-          --container-name=${containerName}
-          --volume-name=${volumeName}`
-      )
-
-      // Output
-      await output.assertSuccess()
-      expect(output.stderr).toContain('Starting PMM Server')
-
-      // http client
-      let client = new PMMRestClient('admin', adminPw, 1080)
-      await client.works()
-
-      // https client
-      client = new PMMRestClient('admin', adminPw, 1443, {
-        baseURL: 'https://localhost:1443',
-        ignoreHTTPSErrors: true,
-      })
-      let resp = await client.doPost('/v1/Settings/Get')
-      let respBody = await resp.json()
-
-      expect(resp.ok()).toBeTruthy()
-      expect(respBody).toHaveProperty('settings')
-
-      // Container name      
-      output = await cli.exec(`docker ps --format="{{.Names}}" | grep "^${containerName}$" | wc -l`)
-      expect(output.stdout.trim()).toEqual('1')
-
-      // Volume name
-      output = await cli.exec(`docker volume ls --format="{{.Name}}" | grep "^${volumeName}$" | wc -l`)
-      expect(output.stdout.trim()).toEqual('1')
-
-      // Docker image
-      output = await cli.exec(`docker ps --format="{{.Names}} {{.Image}}" | grep -E "^${containerName} ${imageName}$" | wc -l`)
-      expect(output.stdout.trim()).toEqual('1')
-    } finally {
-      await teardown([`^${containerName}$`], [volumeName])
-    }
+test.describe('Install PMM Server respects relevant flags', async () => {
+  const adminPw = 'admin123'
+  const containerName = 'pmm-server-install-test'
+  const imageName = 'percona/pmm-server:2.32.0'
+  const volumeName = 'pmm-data-install-test'
+
+  test.beforeAll(async ({}) =>{
+    const output = await cli.exec(`
+      pmm server docker install 
+        --json
+        --admin-password=${adminPw}
+        --docker-image="${imageName}"
+        --https-listen-port=1443
+        --http-listen-port=1080
+        --container-name=${containerName}
+        --volume-name=${volumeName}`
+    )
+    await output.assertSuccess()
+    expect(output.stderr).toContain('Starting PMM Server')
+  });
```

Comment:
Does this approach work with the parallel mode?
When writing the tests I experienced issues with `beforeAll` and the parallel mode since it is executed by every worker.

Reply:
It does, the fluent wait has been missing. But it's not effective. See: https://github.com/percona/pmm/actions/runs/4801870077/jobs/8544628562?pr=1796
The image has been downloaded 5 times. To have an optimization I would ask for a "download" command, so that the image will be downloaded once and the reused for setup without downloading. Or this spec could be switched to chain execution.

Reply:
Can we use project dependencies to run this only once before tests? 
https://playwright.dev/docs/test-projects#dependencies

Reply:
We will to perform "docker pull" to download all images before tests. Though it's limited since setup result is not visible in tests. The official recommended way to pass results to tests via env.VAR which makes limitations and requires some redesign

Reply:
I don't think we need to pull the images. cli pulls the images automatically and we shall test this part as well

Reply:
fixed.